Formula/openjdk.rb

```diff
@@ -30,27 +38,40 @@ class Openjdk < Formula
   # From https://jdk.java.net/archive/
   resource "boot-jdk" do
     on_macos do
-      url "https://download.java.net/java/GA/jdk14.0.2/205943a0976c4ed48cb16f1043c5c647/12/GPL/openjdk-14.0.2_osx-x64_bin.tar.gz"
-      sha256 "386a96eeef63bf94b450809d69ceaa1c9e32a97230e0a120c1b41786b743ae84"
+      if Hardware::CPU.arm?
+        url "https://download.java.net/java/early_access/jdk16/29/GPL/openjdk-16-ea+29_osx-x64_bin.tar.gz"
```

Comment:
Quoting and outstanding task from #65670:

> Before merge, [we'll] need a reliable 16-ea URL or can just hard-link any 16 build if @VladimirKempik thinks it'll be OK.

@vladimirkempik replied with the following information:

> jep391 branch will soon be merged with upstream and become 17-ea

I'm not sure how soon this would happen, but working from upstream (and not a sandbox) is obviously preferred, but we'll need a boot-jdk that won't be pulled down days or weeks after this PR is merged.

@vladimirkempik I'm sorry if I asked this before, but is it absolutely mandatory that the boot-jdk has JEP-391 code?  If not, we can probably find a more reliable link.

If it is mandatory that we have a JEP-391 enabled boot-jdk, I'd some guidance from Homebrew as to where to retrieve this from.  From my experience with other formulae, I understand volatile links are not uncommon, but I don't' know if it's OK here?  Once the bottles are created, it would only affect the source builds, I'm just not sure if that's acceptable.

Reply:
How often would the link break?

Reply:
boot-jdk has to be either jdkX or jdkX-1 ( where X is jdk version you build)
build-jdk has to be same version of jdk you build
both doesn't have to have jep-391 code in it.
so you can just have some fixed version of upstread(jdk/jdk) to be used ad boot/build jdk. this will exist forever

Reply:
> How often would the link break?

It [broke](https://github.com/Homebrew/homebrew-core/pull/65670/commits/3eb4055b7239e4430524b3fa94cb22c6edf4c675) while the old PR was open (26-29, now 31), so they change frequently, I'm not sure how long they maintain the old URLs for though.

Although any JDK16 version will do, I'm not unsure where to find a stable link.   Of the commercial providers, Zulu has an EA link that we could potentially borrow -- just for boot-jdk purposes.

Reply:
I've switched it to the latest (`openjdk-16-ea+31`) and I've mirror it on our side.